crypto/src/test/scala/coop/rchain/crypto/hash/Blake2b512RandomTest.scala

```diff
@@ -43,4 +77,62 @@ class Blake2b512RandomSpec extends FlatSpec with Matchers {
     Base16.encode(res2) should be(
       "438ad6889699497717dd4b05864b0e110888ce906b69944d2d9ca426c4baf9b2")
   }
+  val partialMsg: Array[Byte] = "Hello, World!".getBytes(StandardCharsets.UTF_8)
+  "Partial" should "give a predictable result" in {
+    val b2Random = Blake2b512Random(partialMsg)
+    val res1     = b2Random.next()
+    val res2     = b2Random.next()
+    Base16.encode(res1) should be(
+      "aad51609282445d6552486de7709d9e3abbebe3f521762e1cadb1f5c77fd2ca2")
+    Base16.encode(res2) should be(
+      "de2ea404232f57132cb994fb468f9f3ebf727bab0344f10a54b7a512ab14ba01")
+  }
+  val singleBlockMsg: Array[Byte] =
+    "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa "
+      .getBytes(StandardCharsets.UTF_8)
+  "Single block prefix" should "give a predictable result" in {
+    val b2Random = Blake2b512Random(singleBlockMsg)
+    val res1     = b2Random.next()
+    val res2     = b2Random.next()
+    Base16.encode(res1) should be(
+      "56f3f54bb2fb5064a69db524618535adbc0d69961c3f7f3a21ed172830d10501")
+    Base16.encode(res2) should be(
+      "47f76a84573acfb131f0ef2d20623bd43765ff271a0b2ec24452546679375e00")
+  }
+  val blockAndPartMsg: Array[Byte] =
+    "quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores "
+      .getBytes(StandardCharsets.UTF_8)
+  "Single block and additional partial prefix" should "give a predictable result" in {
+    val b2Random = Blake2b512Random(blockAndPartMsg)
+    val res1     = b2Random.next()
+    val res2     = b2Random.next()
+    Base16.encode(res1) should be(
+      "ba32abd701fe78dd81400ed84b3e6d7e35a5860db79aaa3a8b2f015693bd325f")
+    Base16.encode(res2) should be(
+      "538b649713f6aa8b57971d344b48f27b84058191194fed0dab56741afaae7b74")
+  }
+  val multiBlockMsg: Array[Byte] =
+    "eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem "
+      .getBytes(StandardCharsets.UTF_8)
+  "Multi block prefix" should "give a predictable result" in {
+    val b2Random = Blake2b512Random(multiBlockMsg)
+    val res1     = b2Random.next()
+    val res2     = b2Random.next()
+    Base16.encode(res1) should be(
+      "c6b40c41cb9e3c078936c38bddccce3a017de1e4e751799710c9b50a70feb703")
+    Base16.encode(res2) should be(
+      "b6756c6ebe5e1b2f05ab2212e1a7e8aa126db5637a0111d75715267986df6a95")
+  }
+  val multiBlockAndPartialMsg: Array[Byte] =
```

Comment:
What's the difference between `multiBlockAndPartialMsg`, `multiBlockMsg`, `blockAndPartMsg` , `singleBlockMsg` and `partialMsg` ? Is it the length of the input? They're all tested in the same manner.

Reply:
Yes, it's just the length.
partialMsg is less than a block.
singleBlockMsg is exactly 1 block.
blockAndPartMsg is just over 1 block.
multiblockMsg is 3 (I think) blocks,
multiblockAndPartialMsg is longer than 3 blocks, but not a 4th